app/code/community/Bolt/Boltpay/Block/Catalog/Product/Boltpay.php

```diff
@@ -158,32 +62,38 @@ public function isEnabledProductPageCheckout()
         return ($this->isBoltActive() && $this->boltHelper()->isEnabledProductPageCheckout());
     }
 
-    /**
-     * @return string
-     */
-    public function getProductPageCheckoutSelector()
-    {
-        return $this->boltHelper()->getProductPageCheckoutSelector();
-    }
-
     /**
      * @return bool
      */
     public function isSupportedProductType()
     {
         /** @var Mage_Catalog_Model_Product $product */
-        $product = Mage::registry('current_product');
+        $product = $this->getCurrentProduct();
 
         return ($product && in_array($product->getTypeId(), $this->getProductSupportedTypes()));
     }
 
+    /**
+     * @return object|string
+     */
+    public function getBoltToken(Bolt_Boltpay_Model_BoltOrder $boltOrder, Bolt_Boltpay_Helper_CatalogHelper $helper)
+    {
+        if ($this->isSupportedProductType() && $this->isEnabledProductPageCheckout()) {
+            /** @var Mage_Sales_Model_Quote $ppcQuote */
+            $ppcQuote = $helper->getQuoteWithCurrentProduct($this->getCurrentProduct());
+            $response = $boltOrder->getBoltOrderToken($ppcQuote, Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_MULTI_PAGE);
```

Comment:
hm you shouldn't need order token for product page checkout

Reply:
@daisy1754 in order to show Bolt "Continue Payment" button we need token.
Just look, what will be after you click Bolt button without token? Error.